### PR TITLE
feat(*): add custom sync of user settings

### PR DESCRIPTION
**Pull request by FiboApe**

### CODE REVIEW

File: apps/root/src/state/index.ts [hunk 1]
```diff
@@ -90,8 +93,23 @@ export interface ExtraArgument {
...
 }
 
-const createStore = (web3Service: Web3Service) =>
-  configureStore({
+const persistedSavedActions = [...AggregatorSettingsSavedActions, ...ConfigSavedActions, ...CustomTokensSavedActions];
+
+const createStore = (web3Service: Web3Service) => {
+  const listenerMiddleWare = createListenerMiddleware({ extra: { web3Service } });
+
+  listenerMiddleWare.startListening({
```
Comment (th0rOdinson) [inline apps/root/src/state/index.ts:101]: Fantastic

File: apps/root/src/services/accountService.ts [hunk 2]
```diff
@@ -341,6 +342,12 @@ export default class AccountService extends EventsManager<AccountServiceData> {
         signature: storedSignature,
       };
 
+      const config = accounts[0].config;
+
+      if (config) {
+        this.web3Service.onUpdateConfig(config);
```
Comment (th0rOdinson) [inline apps/root/src/services/accountService.ts:348]: Just to check, isn't empty state an empty JSON? (Current status of all users)

File: apps/root/src/services/meanApiService.ts [hunk 3]
```diff
@@ -29,6 +29,11 @@ import { CLAIM_ABIS } from '@constants/campaigns';
 // MOCKS
 import { getProtocolToken, getWrappedProtocolToken } from '@common/mocks/tokens';
 import { Address, PublicClient, getContract } from 'viem';
+import { SavedCustomConfig } from '@state/base-types';
+
+type AccountWithConfig = Account & {
```
Comment (th0rOdinson) [inline apps/root/src/services/meanApiService.ts:34]: We can move this to a types file

File: apps/root/src/state/aggregator-settings/reducer.ts [hunk 4]
```diff
@@ -56,6 +58,32 @@ export default createReducer(initialState, (builder) => {
     .addCase(setPermit2, (state, { payload }) => {
       state.isPermit2Enabled = payload;
     })
+    .addCase(hydrateAggregatorSettings, (state, { payload }) => {
+      if (!isUndefined(payload.confettiParticleCount)) {
```
Comment (th0rOdinson) [inline apps/root/src/state/aggregator-settings/reducer.ts:62]: Isn't confetti now fixed for everyone? Or you keep this for old users

File: apps/root/src/services/accountService.ts [hunk 5]
```diff
@@ -341,6 +342,12 @@ export default class AccountService extends EventsManager<AccountServiceData> {
...
+      }
```
Comment (th0rOdinson) [inline apps/root/src/services/accountService.ts:349]: What about current config? We would only be storing new changes applied to the store?

File: apps/root/src/services/meanApiService.ts (hunk 3 above)
Comment (FiboApe) [reply apps/root/src/services/meanApiService.ts:34]: Ish, it actually uses a type that is explicitely exclusive to the app, so we don't have access to RootState from the types package :(

File: apps/root/src/state/aggregator-settings/reducer.ts (hunk 4 above)
Comment (FiboApe) [reply apps/root/src/state/aggregator-settings/reducer.ts:62]: Old users + if we later change it and make it configurable again, its already there

**Review by th0rOdinson — APPROVED** [review]

**Pull request merged**